### ISSUE TITLE
TensorFlow integrations can only be built out-of-tree.

### ISSUE THREAD

**Issue by phoenix-meadowlark** [google/iree#4195]
The CMake+Bazel TensorFlow integrations build introduced in https://github.com/google/iree/commit/a3e97f1f74e51524fbd2f2fc27579b8bf15e68f2 can only be built out-of-tree:

```shell
# Fails independently of ccache
cmake -G Ninja -B build/ \
  -DCMAKE_C_COMPILER=clang \
  -DCMAKE_CXX_COMPILER=clang++ \
  -DIREE_ENABLE_CCACHE=ON \
  -DLLVM_CCACHE_BUILD=ON \
  -DIREE_BUILD_TENSORFLOW_COMPILER=ON .
cd build/
ninja
```

```bazel
INFO: Invocation ID: d083be51-7b4c-4c51-a71a-fb96d8bf7b4d
ERROR: infinite symlink expansion detected
[start of symlink chain]
/usr/local/google/home/meadowlark/dev/iree/build/bazel-out/external/iree_core
/usr/local/google/home/meadowlark/dev/iree
[end of symlink chain]
INFO: Repository com_google_protobuf instantiated at:
  no stack (--record_rule_instantiation_callstack not enabled)
Repository rule tf_http_archive defined at:
  /usr/local/google/home/meadowlark/dev/iree/build/bazel-out/external/org_tensorflow/third_party/repo.bzl:131:34: in <toplevel>
ERROR: /usr/local/google/home/meadowlark/dev/iree/integrations/tensorflow/compiler/BUILD:99:10: no such package '@llvm-project//mlir': no such package '@iree_core//': Could not access /usr/local/google/home/meadowlark/dev/iree/build/bazel-out/external/iree_core: Infinite symlink expansion and referenced by '//integrations/tensorflow/compiler:iree-tf-import'
ERROR: /usr/local/google/home/meadowlark/dev/iree/integrations/tensorflow/compiler/BUILD:99:10: no such package '@llvm-project//mlir': no such package '@iree_core//': Could not access /usr/local/google/home/meadowlark/dev/iree/build/bazel-out/external/iree_core: Infinite symlink expansion and referenced by '//integrations/tensorflow/compiler:iree-tf-import'
ERROR: /usr/local/google/home/meadowlark/dev/iree/integrations/tensorflow/compiler/BUILD:99:10: no such package '@llvm-project//mlir': no such package '@iree_core//': Could not access /usr/local/google/home/meadowlark/dev/iree/build/bazel-out/external/iree_core: Infinite symlink expansion and referenced by '//integrations/tensorflow/compiler:iree-tf-import'
ERROR: Analysis of target '//integrations/tensorflow/compiler:iree-tf-import' failed; build aborted: Analysis failed
INFO: Elapsed time: 6.754s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (15 packages loaded, 30 targets configured)
    currently loading: @org_tensorflow//tensorflow/core/platform ... (4 packages)
    Fetching @llvm-project; Restarting.
    Fetching @local_config_cc; Restarting.
    Fetching @local_config_cuda; fetching
    Fetching @local_config_tensorrt; Restarting.
```

Kokoro doesn't give you the permissions to build out-of-tree, so I've implemented a hacky workaround in the kokoro build script for the time being.

**Comment by GMNGeoffrey** [comment]
Kokoro has nothing to do with it really. Docker requires that you mount any filesystems you write to. We currently mount several directories. I think the most logical place to build would be `$HOME/build` which should work fine

**Comment by GMNGeoffrey** [comment]
I'll fix this as part of reworking our Bazel + CMake TF integrations setup

**Comment by GMNGeoffrey** [comment]
Fixed by #4836
